dlib/array/array_kernel.h

```diff
@@ -268,13 +276,29 @@ namespace dlib
     {
         try
         {
-            unsigned long max_size, size;
-            deserialize(max_size,in);
-            deserialize(size,in);
-            item.set_max_size(max_size);
-            item.set_size(size);
-            for (unsigned long i = 0; i < size; ++i)
-                deserialize(item[i],in);
+            unsigned long version;
+            deserialize(version, in);
+
+            if (version != std::numeric_limits<unsigned long>::max()) {
```

Comment:
The max unsigned long value isn't a specific number.  It could be any number of possible values depending on how unsigned long is represented.

More to the point here though, I don't see why you need to edit the serialization code at all.

Reply:
Good point. Changed to 4294967295, but then removed the whole serialization change.